src/main/java/com/github/manolo8/darkbot/utils/CaptchaHandler.java

```diff
@@ -32,16 +33,16 @@ public boolean needsCaptchaSolve(String page) {
         return page.contains("id=\"captchaScriptContainer\"");
     }
 
-    public boolean solveCaptcha() throws IOException {
-        if (setting == null || !setting.getValue() || CaptchaAPI.getInstance() == null) return false;
-        if (isSolvingCaptcha()) return false;
+    public void solveCaptcha() throws IOException {
+        if (setting == null || !setting.getValue() || CaptchaAPI.getInstance() == null) return;
+        if (isSolvingCaptcha()) return;
 
-        System.out.println("CaptchaHandler: Solving Captcha for " + this.action);
-        HttpURLConnection conn = backpage.getHttp(base).getConnection();
-        conn.setInstanceFollowRedirects(false);
-        if (!conn.getHeaderField("Location").isEmpty()) {
-            HttpURLConnection connection = backpage.getHttp(conn.getHeaderField("Location")).getConnection();
-            captchaResponseFuture = CaptchaAPI.getInstance().solveCaptchaFuture(connection.getURL(), IOUtils.read(connection.getInputStream(), true))
+        HttpURLConnection connection = backpage.getHttp(base).getConnection();
+        String page = IOUtils.read(connection.getInputStream());
+
+        if (needsCaptchaSolve(page)) {
```

Comment:
this check can be considered redundant? but makes sure that the url we got has captcha in it.